### PR TITLE
additional docs for windows

### PR DESCRIPTION
**Pull request by klahnakoski**
After ruining my python install I had to fumble to get the tests running on windows again.  This time I recorded those steps.

### CODE REVIEW

File: docs/source/developer.rst [hunk 1]
```diff
@@ -5,24 +5,50 @@ Fastparquet is a free and open-source project.
...
+Running tests
+-------------
 
 Aside from the requirements for using this package, the following additional
 packages should be present:
 
-- pytest
+.. code-block:: bash
+
+    pip install pytest
```
Comment (martindurant) [inline docs/source/developer.rst]: conda?

File: docs/source/developer.rst [hunk 2]
```diff
@@ -5,24 +5,50 @@ Fastparquet is a free and open-source project.
...
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `python-snappy` is required for tests, but it requires a compiler,
+which can be difficult to get running on Windows. Instead, `download an
+unofficial pre-compiled version <https://www.lfd.uci.edu/~gohlke/pythonlibs/#python-snappy>`_
+and install.
+
+.. code-block:: bash
+
+    pip install python_snappy-0.5.1-cp27-cp27m-win32.whl
```
Comment (martindurant) [inline docs/source/developer.rst]: Did you not use/succeed with conda?

**Comment by klahnakoski** [comment]
"Conda"?  You mean this: https://conda.io/docs/user-guide/install/index.html ?

The `README.rst` implies installation for use, not installation for development.  Even so, I would think the `pypi` install would work.

I do admit to only one train of thought:  Fork the repo, run setup command(s), then run tests. If I must make choices during installation, I am going to screw it up.

**Comment by martindurant** [comment]
Conda is a package management system, like a more advanced virtualenv - you generally have a much easier time getting dependencies and do not have to worry about compilation (all installs are binary). Naturally I am biased as an employee of Anaconda, but I was using conda before and would do now regardless.

**Comment by klahnakoski** [comment]
Conda is yet-another-package-manager?!  Python is hard mostly because there are too many package managers.  I see now that conda is not just one package manager, [rather two-more-package-managers](https://conda.io/docs/user-guide/install/windows.html).  Which one should I be using for installing `fastparquet` in development mode?

**Comment by martindurant** [comment]
This is perhaps not the place to discuss one's favourite development environment, but I can encourage to read the following on the subject of conda: http://jakevdp.github.io/blog/2016/08/25/conda-myths-and-misconceptions/
For the interested, the definition to build the conda package for fastparquet is [here](https://github.com/conda-forge/fastparquet-feedstock/blob/master/recipe/meta.yaml).

As a new developer, I would probably conda-install the package itself, or explicitly install the dependencies (which is why the instructions say "get" without saying "how"), perhaps in a fresh environment, before doing git-clone, python-setup.py on a local copy of the code for development.

**Comment by martindurant** [comment]
(to specifically answer you question, you get conda with either "miniconda" or "anaconda", the difference being that the former gives you the bare minimum so that you choose what else to install, and the latter gives you a batteries-included set of packages for data science)

**Comment by klahnakoski** [comment]
Ok, I will go through the install sequence with miniconda and get back to you...

**Comment by klahnakoski** [comment]
I installed miniconda, and then install fastparquet:

```
c:\Users\kyle\code>C:\ProgramData\Miniconda2\Scripts\conda.exe install -c conda-forge fastparquet
Fetching package metadata ...............
Solving package specifications: .

Package plan for installation in environment C:\ProgramData\Miniconda2:

The following NEW packages will be INSTALLED:

    attrs:           17.4.0-py_0                conda-forge
    colorama:        0.3.9-py27_0               conda-forge
    cython:          0.27.3-py27_0              conda-forge
    fastparquet:     0.1.3-py27_0               conda-forge
    funcsigs:        1.0.2-py_2                 conda-forge
    icc_rt:          2017.0.4-h97af966_0
    intel-openmp:    2018.0.0-hd92c6cd_8
    llvmlite:        0.21.0-py27h831ec56_0
    mkl:             2018.0.1-h2108138_4
    numba:           0.36.2-np113py27hb75b33f_0
    numpy:           1.13.3-py27h9ac254c_0
    pandas:          0.22.0-py27_0              conda-forge
    pluggy:          0.6.0-py_0                 conda-forge
    py:              1.5.2-py_0                 conda-forge
    pytest:          3.3.2-py27_0               conda-forge
    python-dateutil: 2.6.1-py27_0               conda-forge
    pytz:            2017.3-py_2                conda-forge
    singledispatch:  3.4.0.3-py27_0             conda-forge
    thrift:          0.10.0-py27_0              conda-forge
    vs2015_runtime:  14.0.25420-0               conda-forge

The following packages will be SUPERSEDED by a higher-priority channel:

    conda:           4.3.31-py27_0                          --> 4.3.31-py27_0 conda-forge
    conda-env:       2.6.0-h36134e3_1                       --> 2.6.0-0       conda-forge
    openssl:         1.0.2n-hc64555f_0                      --> 1.0.2n-vc9_0  conda-forge [vc9]
    yaml:            0.1.7-h3e6d941_2                       --> 0.1.7-vc9_0   conda-forge [vc9]

Proceed ([y]/n)? y

conda-env-2.6. 100% |###############################| Time: 0:00:00  44.91 kB/s
icc_rt-2017.0. 100% |###############################| Time: 0:00:03   2.49 MB/s
intel-openmp-2 100% |###############################| Time: 0:00:00   2.40 MB/s
vs2015_runtime 100% |###############################| Time: 0:00:01   1.12 MB/s
mkl-2018.0.1-h 100% |###############################| Time: 0:01:07   2.42 MB/s
openssl-1.0.2n 100% |###############################| Time: 0:00:02   1.66 MB/s
yaml-0.1.7-vc9 100% |###############################| Time: 0:00:00 529.38 kB/s
attrs-17.4.0-p 100% |###############################| Time: 0:00:00 508.79 kB/s
colorama-0.3.9 100% |###############################| Time: 0:00:00  35.28 kB/s
cython-0.27.3- 100% |###############################| Time: 0:00:02 910.91 kB/s
funcsigs-1.0.2 100% |###############################| Time: 0:00:00 919.81 kB/s
numpy-1.13.3-p 100% |###############################| Time: 0:00:01   2.74 MB/s
pluggy-0.6.0-p 100% |###############################| Time: 0:00:00 275.19 kB/s
py-1.5.2-py_0. 100% |###############################| Time: 0:00:00 655.92 kB/s
pytz-2017.3-py 100% |###############################| Time: 0:00:00 899.43 kB/s
llvmlite-0.21. 100% |###############################| Time: 0:00:03   2.47 MB/s
python-dateuti 100% |###############################| Time: 0:00:00 883.69 kB/s
singledispatch 100% |###############################| Time: 0:00:00   1.05 MB/s
thrift-0.10.0- 100% |###############################| Time: 0:00:00   1.54 MB/s
numba-0.36.2-n 100% |###############################| Time: 0:00:00   2.46 MB/s
pandas-0.22.0- 100% |###############################| Time: 0:00:05   1.70 MB/s
pytest-3.3.2-p 100% |###############################| Time: 0:00:00   1.30 MB/s
fastparquet-0. 100% |###############################| Time: 0:00:00   1.11 MB/s
conda-4.3.31-p 100% |###############################| Time: 0:00:00   1.22 MB/s
```

I cloned the fastparquet repo:

```
C:\Users\kyle\code>git clone https://github.com/dask/fastparquet.git fastparquet-conda
Cloning into 'fastparquet-conda'...
remote: Counting objects: 2799, done.
remote: Compressing objects: 100% (23/23), done.
remote: Total 2799 (delta 5), reused 11 (delta 4), pack-reused 2771
Receiving objects: 100% (2799/2799), 27.41 MiB | 862.00 KiB/s, done.
Resolving deltas: 100% (1966/1966), done.
Checking connectivity... done.
```

and then ran tests 

```
C:\Users\kyle\code>cd fastparquet-conda

C:\Users\kyle\code\fastparquet-conda>C:\ProgramData\Miniconda2\Scripts\py.test.exe fastparquet/test/test_api.py
============================= test session starts =============================
platform win32 -- Python 2.7.14, pytest-3.3.2, py-1.5.2, pluggy-0.6.0
rootdir: C:\Users\kyle\code\fastparquet-conda, inifile:
collected 0 items / 1 errors

=================================== ERRORS ====================================
________________ ERROR collecting fastparquet/test/test_api.py ________________
ImportError while importing test module 'C:\Users\kyle\code\fastparquet-conda\fastparquet\test\test_api.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
fastparquet\__init__.py:8: in <module>
    from .core import read_thrift
fastparquet\core.py:13: in <module>
    from . import encoding
fastparquet\encoding.py:11: in <module>
    from .speedups import unpack_byte_array
E   ImportError: No module named speedups
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!
=========================== 1 error in 5.73 seconds ===========================
```

It seems the conda install has the same problem: The speedups are not compiled. I compile the speedups:

```
C:\Users\kyle\code\fastparquet-conda>c:\ProgramData\Miniconda2\python.exe setup.py build_ext --inplace
```
but the Snappy failures still happen

```
fastparquet\test\test_with_n.py ......F                                  [100%]

================================== FAILURES ===================================
__________________ test_roundtrip[SNAPPY-row_groups0-simple] __________________

tempdir = 'c:\\users\\kyle\\appdata\\local\\temp\\tmptaqoth', scheme = 'simple'
row_groups = [0], comp = 'SNAPPY'

    @pytest.mark.parametrize('scheme', ['simple', 'hive'])
    @pytest.mark.parametrize('row_groups', [[0], [0, 500]])
    @pytest.mark.parametrize('comp', [None, 'GZIP', 'SNAPPY'])
    def test_roundtrip(tempdir, scheme, row_groups, comp):
        data = pd.DataFrame({'i32': np.arange(1000, dtype=np.int32),
                             'i64': np.arange(1000, dtype=np.int64),
                             'u64': np.arange(1000, dtype=np.uint64),
                             'f': np.arange(1000, dtype=np.float64),
                             'bhello': np.random.choice([b'hello', b'you',
                                b'people'], size=1000).astype("O")})
        data['a'] = np.array([b'a', b'b', b'c', b'd', b'e']*200, dtype="S1")
        data['aa'] = data['a'].map(lambda x: 2*x).astype("S2")
        data['hello'] = data.bhello.str.decode('utf8')
        data['bcat'] = data.bhello.astype('category')
        data['cat'] = data.hello.astype('category')
        fname = os.path.join(tempdir, 'test.parquet')
        write(fname, data, file_scheme=scheme, row_group_offsets=row_groups,
>             compression=comp)

fastparquet\test\test_output.py:113:
```
and then install `python-snappy`:

```
C:\Users\kyle\code\fastparquet-conda>C:\ProgramData\Miniconda2\Scripts\pip.exe install C:\Users\kyle\code\fastparquet\resources\python_snappy-0.5.1-cp27-cp27m-win_amd64.whl
Processing c:\users\kyle\code\fastparquet\resources\python_snappy-0.5.1-cp27-cp27m-win_amd64.whl
Installing collected packages: python-snappy
Successfully installed python-snappy-0.5.1
```

Then all tests run:

```
C:\Users\kyle\code\fastparquet-conda>C:\ProgramData\Miniconda2\Scripts\py.test.exe fastparquet/test
============================= test session starts =============================
platform win32 -- Python 2.7.14, pytest-3.3.2, py-1.5.2, pluggy-0.6.0
rootdir: C:\Users\kyle\code\fastparquet-conda, inifile:
collected 209 items

fastparquet\test\test_api.py .......................                     [ 11%]
fastparquet\test\test_aroundtrips.py sssssssssssssssssssssssssss         [ 23%]
fastparquet\test\test_compression.py .....                               [ 26%]
fastparquet\test\test_converted_types.py ..s...s....                     [ 31%]
fastparquet\test\test_dataframe.py .                                     [ 32%]
fastparquet\test\test_encoding.py ............                           [ 37%]
fastparquet\test\test_output.py ....s........................s.......... [ 56%]
..................s.................                                     [ 74%]
fastparquet\test\test_partition_filters_specialstrings.py ssssssssss     [ 78%]
fastparquet\test\test_read.py ......s..................                  [ 90%]
fastparquet\test\test_speedups.py ....                                   [ 92%]
fastparquet\test\test_thrift_structures.py ..                            [ 93%]
fastparquet\test\test_util.py ......                                     [ 96%]
fastparquet\test\test_with_n.py .......                                  [100%]

=================== 166 passed, 43 skipped in 49.38 seconds ===================
```

**Comment by martindurant** [comment]
Yep, that sounds like the right sequence of actions. You can also create an independent py36 environment from the same conda install for testing if you wish.

**Comment by klahnakoski** [comment]
what is the command to "create an independent py36 environment from the same conda install"?

**Comment by martindurant** [comment]
```
conda create -n py36 python=3.6 -c conda-forge fastparquet dask ...
```
(-c conda-forge: prefer packages supplied by conda-forge
-n mynewenv: name of environment
add additional packages as required
to "use" the environment, you need `source activate py36`)

**Comment by martindurant** [comment]
From the point of view of this PR, I would either give full instructions for conda, and pip, or leave things as they were.

**Comment by klahnakoski** [comment]
ok, we will close this then

**Pull request closed without merging**